internal/common/rules/globalrulescache.go

```diff
@@ -11,52 +10,103 @@ var AvailableRules = map[string]RuleMatrix{}
 // RulesInPlay is a global cache of all rules currently in effect
 var RulesInPlay = map[string]RuleMatrix{}
 
+// A RuleError is a non-critical issue which can be caused by an island trying to modify, register or pick rules which isn't mechanically feasible
+type RuleError int
```

Comment:
https://www.digitalocean.com/community/tutorials/creating-custom-errors-in-go

Reply:
https://golangbot.com/custom-errors/ <- this is clearer.

Reply:
You can keep this iota, but you can define your new error, say
```
type RuleErrorType int // <- this is your iota
type RuleError struct {
    err error
    type RuleError
}

func (e *RuleError) Error() string {
    return e.err.Error()
}

func (e *RuleError) Type() RuleErrorType {
    return e.Type
} 
```

Usage:
```
func SomeRuleFunc(...) error {
    return &RuleError { err: errors.Errorf("REEEE"), type: RuleIsNotInPlay }
}
err := SomeRuleFunc()
if err != nil {
    if ruleErr, ok := err.(*RuleError); ok {
        fmt.Println("Rule error of type '%v'", ruleErr.Type())
    } else {
        // handle like a normal error
    }
}
```

Reply:
also see #150 for iota implementations. just a copy a paste thing. thanks.

Reply:
Thanks for this, this is a neat approach, will implemenent

Reply:
Ok added, pls have another look